### PR TITLE
HA: Set Mandatory ordering for Heat startup

### PR DESCRIPTION
**Pull request by dirkmueller**
Heat depends on Nova, Keystone and friends to be started first.

### CODE REVIEW

**Comment by vuntz** [comment]
See comments in https://github.com/crowbar/barclamp-keystone/pull/267

I also don't believe that heat needs nova: you can list stack with nova being down.

**Comment by aspiers** [comment]
This requires https://github.com/crowbar/barclamp-pacemaker/pull/179 so should not be merged until that one is.

File: chef/cookbooks/heat/recipes/ha.rb [hunk 1]
```diff
@@ -72,4 +72,11 @@
   only_if { CrowbarPacemakerHelper.is_cluster_founder?(node) }
 end
 
+crowbar_pacemaker_order_only_existing "o-#{group_name}" do
+  ordering "rabbitmq cl-keystone cl-g-nova-controller cl-#{group_name}"
+  score "Mandatory"
+  action [ :create ]
```
Comment (aspiers) [inline chef/cookbooks/heat/recipes/ha.rb]: No need for the `[ ]` here, IIRC.

**Comment by aspiers** [comment]
+1 except for the `[ ]` nitpick and the waiting for https://github.com/crowbar/barclamp-pacemaker/pull/179 to be merged.

**Comment by aspiers** [comment]
Are we targeting this for GM?  Would require proper QA.

**Comment by vuntz** [comment]
Submitted https://github.com/dirkmueller/barclamp-heat/pull/1 for my fixes

File: chef/cookbooks/heat/recipes/ha.rb [hunk 2]
```diff
@@ -72,4 +72,11 @@
   only_if { CrowbarPacemakerHelper.is_cluster_founder?(node) }
 end
 
+crowbar_pacemaker_order_only_existing "o-#{group_name}" do
+  ordering "rabbitmq cl-keystone cl-g-nova-controller cl-#{group_name}"
```
Comment (vuntz) [inline chef/cookbooks/heat/recipes/ha.rb]: Any reason you skipped postgresql here?

**Comment by dirkmueller** [comment]
@vuntz thats true, but it makes no sense to start heat before nova is up, and I've seen exceptions in heat-engine if nova is unavailable already - so its better to order the startup.

**Comment by vuntz** [comment]
+1

**Pull request merged**